### DISCUSSION TITLE
What about my API documentation now that Swashbuckle (Swagger) is no longer a dependency in ASP.NET 9

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#378] [Announcements]
# What about my API documentation now that Swashbuckle (Swagger) is no longer a dependency in ASP.NET 9

As you will see in this blog post, the "baked-in" Swashbuckle package is no longer a dependency in the ASP.NET 9 project template. Let's take a look at what this means for your API documentation and go through some options to render the OpenAPI document.

https://timdeschryver.dev/blog/what-about-my-api-documentation-now-that-swashbuckle-is-no-longer-a-dependency-in-aspnet-9

**Comment by dodyg** [comment]
The NSWag example will fail.

**Reply by timdeschryver** [reply]
Thanks for mentioning it. I will try to update the example in a couple of days.
Do you know/remember why it fails?

**Comment by DavidRogersDev** [comment]
Where does [Asp.Versioning.Mvc.ApiExplorer](https://www.nuget.org/packages/Asp.Versioning.Mvc.ApiExplorer) fit in with this stuff? Is that the package which scrapes the bits from the relevant attributes in Controllers to create the OpenAPI json document?

**Reply by timdeschryver** [reply]
@DavidRogersDev I don't believe so. The library helps you to build versioning into your API, see the wiki for more info at https://github.com/dotnet/aspnet-api-versioning/wiki.

The `Microsoft.AspNetCore.OpenApi` creates the OpenAPI document based on the attributes in controllers, or methods with Minimal APIs.
